src/common.rs

```diff
@@ -119,3 +119,39 @@ macro_rules! impl_extract {
 }
 
 pub(crate) use impl_extract;
+
+macro_rules! impl_step {
+    ($type:tt as $num_trait:tt) => {
```

Comment:
num_trait is an optional trait dependency. Maybe I'm reading this wrong, but why does this pass tests?

Reply:
I forgot there's support for that crate, knowing that I realize it's a bit of a confusing variable name 😅 

The user of the macro is meant to pass in `Number`/`SignedNumber` so that the macro can generate the correct trait bounds, so I figured I'd call the argument something along the lines of "number trait". I'll change it to something more distinct.